recipes/quickstart/NotebookLlama/Step-1 PDF-Pre-Processing-Logic.ipynb

```diff
@@ -0,0 +1,2731 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "4f67a6a6",
+   "metadata": {},
+   "source": [
+    "## Notebook 1: PDF Pre-processing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f68aee84-04e3-4cbc-be78-6de9e06e704f",
+   "metadata": {},
+   "source": [
+    "In the series, we will be going from a PDF to Podcast using all open models. \n",
+    "\n",
+    "The first step in getting to the podcast is finding a script, right now our logic is:\n",
+    "- Use any PDF on any topic\n",
+    "- Prompt `Llama-3.2-1B-Instruct` model to process it into a text file\n",
+    "- Re-write this into a podcast transcript in next notebook.\n",
+    "\n",
+    "In this notebook, we will upload a PDF and save it into a `.txt` file using the `PyPDF2` library, later we will process chunks from the text file using our featherlight model."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "61cb3584",
+   "metadata": {},
+   "source": [
+    "Most of us shift-enter pass the comments to realise later we need to install libraries. For the few that read the instructions, please remember to do so:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "id": "f4fc7aef-3505-482e-a998-790b8b9d48e4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#!pip install PyPDF2\n",
+    "#!pip install rich ipywidgets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b23d509",
+   "metadata": {},
+   "source": [
+    "Assuming you have a PDF uploaded on the same machine, please set the path for the file. \n",
+    "\n",
+    "Also, if you want to flex your GPU-please switch to a bigger model although the featherlight models work perfectly for this task:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "60d0061b-8b8c-4353-850f-f19466a0ae2d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pdf_path = './2402.13116v3.pdf'\n",
+    "DEFAULT_MODEL = \"meta-llama/Llama-3.2-1B-Instruct\""
```

Comment:
do we need to set the HF token?

Reply:
Sorry missed this-I'll add instructions in the ReadMe to login using the HF CLI